api/envoy/extensions/filters/http/dynamic_modules/v3/dynamic_modules.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.dynamic_modules.v3;
+
+import "envoy/extensions/dynamic_modules/v3/dynamic_modules.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.dynamic_modules.v3";
+option java_outer_classname = "DynamicModulesProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/dynamic_modules/v3;dynamic_modulesv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: HTTP filter for dynamic modules]
+
+// Configuration of the HTTP filter for dynamic modules. This filter allows loading shared object files
+// that can be loaded via dlopen by the HTTP filter.
+//
+// A module can be loaded by multiple HTTP filters, hence the program can be structured in a way that
+// the module is loaded only once and shared across multiple filters providing multiple functionalities.
+//
+// Currently, the implementation is work in progress and not usable.
+message DynamicModuleFilter {
+  // Specifies the shared-object level configuration.
+  envoy.extensions.dynamic_modules.v3.DynamicModuleConfig dynamic_module_config = 1;
+
+  // The name for this filter configuration. This can be used to distinguish between different filter implementations
+  // inside a dynamic module. For example, a module can have completely different filter implementations.
+  // At the start up of a filter chain, this name is passed to the module's HTTP filter initialization function.
+  // That way the module can decide which in-module filter implementation to use based on the name.
+  string filter_name = 2;
```

Comment:
Per the other comment this should be done at configuration load/validation time, not when the filter chain is created.

Reply:
Also I'm not sure `filter_name` is a good name here if we eventually support loadable modules for non filter extensions. Perhaps something like `fully_qualified_extension_name` or something. To that end, I wonder if this should actually go inside the shared `DynamicModuleConfig` ? Then that config could de-dup what to load based on the module name, but still have the "extension name" in that structure.

Reply:
Sorry my documentation was misleading and wrong. This is intended to be done at the load/validation time of HTTP filter configuration (when it creates a factory). 

As for the naming and placement, from implementation perspective, the initialization function will differ among different extension points (`envoy_dynamic_modules_http_filter_config_new`, `envoy_dynamic_modules_tcp_filter_config_new`, `envoy_dynamic_modules_access_logging_new`, etc), so we would need a "typed" `fully_qualified_extension_name` which has a info about which extension point being used. I think then it would be simpler to put this field in an extension specific proto (this case http/dynamic_modules.proto) instead of putting this in a common proto.

Reply:
https://github.com/envoyproxy/envoy/pull/36448/commits/5a953a9cecb987539021a259099afc9fccdfc5ee